webapp/src/hooks/customHooks/useSpendToolsState.js

```diff
@@ -0,0 +1,179 @@
+  const [openSnackbar, setOpenSnackbar] = useState(false)
+  const [amountClaimed, setAmountClaimed] = useState(0)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [openModal, setOpenModal] = useState(false)
+  const [modalData, setModalData] = useState({})
+
+  const [formValues, handleInputChange, reset, errors, validateForm] = useForm({
+    to: '',
+    amount: '',
+    category: '',
+    description: ''
+  })
+  const [
+    formValuesModal,
+    handleInputChangeModal,
+    resetModal,
+    errorsModal,
+    validateFormModal
+  ] = useForm({
+    newCategory: '',
+    newDescription: ''
+  })
+
+  const executeAction = async (data, account, name) => {
+    setErrorMessage('')
+    const transaction = {
+      actions: [
+        {
+          authorization: [
+            {
+              actor: state.user?.accountName,
+              permission: 'active'
+            }
+          ],
+          account,
+          name,
+          data
+        }
+      ]
+    }
+
+    try {
+      await state.ual.activeUser.signTransaction(transaction, {
+        broadcast: true
+      })
+
+      setOpenSnackbar(true)
+
+      openModal ? resetModal() : reset()
+    } catch (error) {
+      setErrorMessage(error.message)
+    }
+  }
+
+  const handleCloseModal = () => {
+    setOpenModal(false)
+    setErrorMessage('')
+  }
+
+  const handleOpenModal = transaction => {
+    setModalData(transaction.row)
+    setOpenModal(true)
+    setErrorMessage('')
+  }
+
+  const [loadUncategorizedTransactions, { data: uncategorizedTxData }] =
+    useLazyQuery(GET_UNCATEGORIZED_TRANSACTIONS_BY_ACCOUNT_QUERY, {
+      variables: {
+        account: authenticatedUser
+      }
+    })
+
+  const [loadCategorizedAmount, { data: categorizedTxData }] = useLazyQuery(
+    GET_CATEGORIZED_TRANSACTIONS_BY_ACCOUNT_QUERY,
+    {
+      variables: {
+        account: authenticatedUser,
+        category: 'uncategorized',
+        election: currentElection
+      }
+    }
+  )
+
+  const [loadClaimedAmount, { data: claimedTxData }] = useLazyQuery(
+    GET_CATEGORIZED_TRANSACTIONS_BY_ACCOUNT_QUERY,
+    {
+      variables: {
+        account: authenticatedUser,
+        category: 'unclaimed',
+        election: currentElection
+      }
+    }
+  )
+
+  const [loadCurrentElection, { data: currentElectionData }] = useLazyQuery(
+    GET_ACTUAL_ELECTION_QUERY
+  )
+
+  useEffect(() => {
+    loadUncategorizedTransactions()
+    loadCategorizedAmount()
+    loadClaimedAmount()
+    loadCurrentElection()
+  }, [])
+
+  useEffect(() => {
+    setTransactionsList(uncategorizedTxData?.eden_transaction || [])
+  }, [uncategorizedTxData])
+
+  useEffect(() => {
+    setAmountCategorized(
+      categorizedTxData?.eden_transaction_aggregate?.aggregate?.sum?.amount || 0
+    )
+  }, [categorizedTxData])
+
+  useEffect(() => {
+    setAmountClaimed(
+      claimedTxData?.eden_transaction_aggregate?.aggregate?.sum?.amount || 0
+    )
+  }, [categorizedTxData])
```

Comment:
Why this useEffect has `categorizedTxData` as a dependency but never used?

the dependency should be `claimedTxData`